api/adapters/hydroshare.py

```diff
@@ -0,0 +1,260 @@
+import requests
+from starlette import status
+from datetime import datetime
+from typing import List, Optional, Union
+from pydantic import BaseModel, EmailStr, HttpUrl
+from api.adapters.base import AbstractRepositoryMetadataAdapter, AbstractRepositoryRequestHandler
+from api.exceptions import RepositoryException
+from api.models import schema
+from api.models.catalog import DatasetMetadataDOC
+
+
+class _HydroshareRequestHandler(AbstractRepositoryRequestHandler):
+
+    def get_metadata(self, record_id: str):
+        hs_meta_url = self.settings.hydroshare_meta_read_url % record_id
+        hs_file_url = self.settings.hydroshare_file_read_url % record_id
+
+        def make_request(url, file_list=False):
+            response = requests.get(url)
+            if response.status_code != status.HTTP_200_OK:
+                raise RepositoryException(status_code=response.status_code, detail=response.text)
+            if not file_list:
+                return response.json()
+
+            content_files = []
+            content_files.extend(response.json()["results"])
+            # check if there are more results to fetch - by default, 100 files are returned from HydroShare
+            while response.json()["next"]:
+                response = requests.get(response.json()["next"])
+                if response.status_code != status.HTTP_200_OK:
+                    raise RepositoryException(status_code=response.status_code, detail=response.text)
+                content_files.extend(response.json()["results"])
+            return content_files
+
+        metadata = make_request(hs_meta_url)
+        files_metadata = make_request(hs_file_url, file_list=True)
+        metadata["content_files"] = files_metadata
+        return metadata
+
+
+class HydroshareMetadataAdapter(AbstractRepositoryMetadataAdapter):
+    repo_api_handler = _HydroshareRequestHandler()
+
+    @staticmethod
+    def to_catalog_record(metadata: dict) -> DatasetMetadataDOC:
+        """Converts hydroshare resource metadata to a catalog dataset record"""
+        hs_metadata_model = _HydroshareResourceMetadata(**metadata)
```

Comment:
The schema can include aggregation and file metadata.  Should we include that metadata as well?

Reply:
I think I have once discussed aggregation/file metadata with Jeff Horsburgh as part of resource registration work, and he suggested that we may consider that in a future version.